hack/dump_scenario_from_k8s.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+set -e
+
+readonly ARGS=("$@")
+
+usage() {
+    cat <<-EOF
+    usage: dump_scenario_from_k8s.sh options
+
+    Dump the current scenario from a running trento-server installation on the k8s cluster
+
+    OPTIONS:
+        -n, --name            The name to use for the scenario. Defaults to "current".
+        -p, --path            The path where the scenario should be saved. Defaults to the current directory.
+        -h, --help            Print this help.
+
+    Example:
+        dump_scenario_from_k8s.sh --name failover --path /tmp
+EOF
+}
+
+cmdline() {
+    local arg=
+
+    for arg; do
+        local delim=""
+        case "$arg" in
+        --name) args="${args}-n " ;;
+        --path) args="${args}-p " ;;
+        --help) args="${args}-h " ;;
+
+        # pass through anything else
+        *)
+            [[ "${arg:0:1}" == "-" ]] || delim="\""
+            args="${args}${delim}${arg}${delim} "
+            ;;
+        esac
+    done
+
+    eval set -- "$args"
+
+    while getopts "n:p:h" OPTION; do
+        case $OPTION in
+        h)
+            usage
+            exit 0
+            ;;
+        n)
+            readonly NAME=$OPTARG
+            ;;
+        p)
+            readonly EXPORT_PATH=$OPTARG
+            ;;
+        *)
+            usage
+            exit 0
+            ;;
+        esac
+    done
+
+    return 0
+}
+
+dump-scenario() {
+    local name=${NAME:-current}
+    local path="${EXPORT_PATH:-$PWD}"
+
+    if [[ -d "$path/scenarios/$name" ]]; then
+        echo "The scenario $name already exists in $path/scenarios/$name"
+        echo "Please choose a different name."
+        exit 1
+    fi
+
+    kubectl exec -ti deploy/trento-server-web -- /app/trento ctl dump-scenario --name="$name" --path /scenarios
+    kubectl exec deploy/trento-server-web -- tar cf - scenarios | tar xf - -C "$path"
```

Comment:
Where is this creating the tarball?
To my eyes it is created in the pod itself. Or how is it passed to the local machine?
Or not using the `-ti` flags does the magic for this?

Reply:
This is not creating a tarball, but it is the suggested way to copy a file over and it is using tar to copy the whole dir.
https://kubernetes.io/docs/reference/kubectl/cheatsheet/#copy-files-and-directories-to-and-from-containers
It is the equivalent of  `kubectl cp` but I need to use the `exec` instead because `cp` doesn't allow to specify a deployment but just a container/pod.

Reply:
Thanks! Understood